src/rules/property-no-unknown/index.js

```diff
@@ -0,0 +1,75 @@
+import {
+  isString,
+  isArray,
+  isBoolean,
+} from "lodash"
+import {
+  isStandardSyntaxProperty,
+  isCustomProperty,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+import { all as properties } from "known-css-properties"
+
+export const ruleName = "property-no-unknown"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (property) => `Unexpected unknown property "${property}"`,
+})
+
+const isPropertyIgnored = (prop, ignore) => {
+  if (isArray(ignore)) {
+    return ignore.indexOf(prop) > -1
+  }
+
+  if (isString(ignore)) {
+    return prop === ignore
+  }
+
+  return false
+}
+
+const isPropertyValid = (prop) => {
+  return properties.indexOf(prop) > -1
+}
+
+const validate = (result, options) => {
+  return function (node) {
+    const prop = node.prop
```

Comment:
Idiomatically, we use destructuring to access properties e.g. `const { prop } = node`